tests/TestCase/Database/Expression/FunctionExpressionTest.php

```diff
@@ -96,6 +97,45 @@ public function testFunctionNestingQueryExpression()
         $this->assertEquals('MyFunction(a)', $f->sql($binder));
     }
 
+    /**
+     * Tests that passing a database query as an argument wraps the query SQL into parentheses.
+     *
+     * @return void
+     */
+    public function testFunctionWithDatabaseQuery()
+    {
+        $query = ConnectionManager::get('test')
+            ->newQuery()
+            ->select(['column']);
+
+        $binder = new ValueBinder;
+        $function = new FunctionExpression('MyFunction', [$query]);
+        $this->assertEquals(
+            'MyFunction((SELECT column))',
+            preg_replace('/[`"\[\]]/', '', $function->sql($binder))
+        );
+    }
+
+    /**
+     * Tests that passing a ORM query as an argument wraps the query SQL into parentheses.
+     *
+     * @return void
+     */
+    public function testFunctionWithOrmQuery()
+    {
+        $query = $this->getTableLocator()->get('Articles')
+            ->setSchema(['column' => 'integer'])
+            ->find()
+            ->select(['column']);
+
+        $binder = new ValueBinder;
+        $function = new FunctionExpression('MyFunction', [$query]);
+        $this->assertEquals(
+            'MyFunction((SELECT Articles.column AS Articles__column FROM articles Articles))',
+            preg_replace('/[`"\[\]]/', '', $function->sql($binder))
+        );
```

Comment:
It would be good to have a test that hits the database as well to ensure that we aren't introducing incompatible syntax with any of our supported DB platforms.

Reply:
I agree, we should have several functional test for this change. I having something like this in the early days of QueryExpression, and it turned out to be a problem because some database servers did not like the extra parens

Reply:
Sure, I can add tests for that, but I'm pretty positive that this is not only valid, but required in all supported dialects, even in unsupported ones like Oracle or Firebase.